internal/store/json.go

```diff
@@ -0,0 +1,303 @@
+	Blueprints blueprintsV0 `json:"blueprints"`
+	Workspace  workspaceV0  `json:"workspace"`
+	Composes   composesV0   `json:"composes"`
+	Sources    sourcesV0    `json:"sources"`
+	Changes    changesV0    `json:"changes"`
+	Commits    commitsV0    `json:"commits"`
+}
+
+type blueprintsV0 map[string]blueprint.Blueprint
+type workspaceV0 map[string]blueprint.Blueprint
+
+// A Compose represent the task of building a set of images from a single blueprint.
+// It contains all the information necessary to generate the inputs for the job, as
+// well as the job's state.
+type composeV0 struct {
+	Blueprint   *blueprint.Blueprint `json:"blueprint"`
+	ImageBuilds []imageBuildV0       `json:"image_builds"`
+}
+
+type composesV0 map[uuid.UUID]composeV0
+
+// ImageBuild represents a single image build inside a compose
+type imageBuildV0 struct {
+	ID          int               `json:"id"`
+	ImageType   common.ImageType  `json:"image_type"`
+	Manifest    *osbuild.Manifest `json:"manifest"`
+	Targets     []*target.Target  `json:"targets"`
+	JobCreated  time.Time         `json:"job_created"`
+	JobStarted  time.Time         `json:"job_started"`
+	JobFinished time.Time         `json:"job_finished"`
+	Size        uint64            `json:"size"`
+	JobID       uuid.UUID         `json:"jobid,omitempty"`
+
+	// Kept for backwards compatibility. Image builds which were done
+	// before the move to the job queue use this to store whether they
+	// finished successfully.
+	QueueStatus common.ImageBuildState `json:"queue_status,omitempty"`
+}
+
+type sourceV0 struct {
+	Name     string `json:"name"`
+	Type     string `json:"type"`
+	URL      string `json:"url"`
+	CheckGPG bool   `json:"check_gpg"`
+	CheckSSL bool   `json:"check_ssl"`
+	System   bool   `json:"system"`
+}
+
+type sourcesV0 map[string]sourceV0
+
+type changeV0 struct {
+	Commit    string `json:"commit"`
+	Message   string `json:"message"`
+	Revision  *int   `json:"revision"`
+	Timestamp string `json:"timestamp"`
+}
+
+type changesV0 map[string]map[string]changeV0
+
+type commitsV0 map[string][]string
+
+func newBlueprintsFromV0(blueprintsStruct blueprintsV0) map[string]blueprint.Blueprint {
+	blueprints := make(map[string]blueprint.Blueprint)
+	for name, blueprint := range blueprintsStruct {
+		blueprints[name] = blueprint.DeepCopy()
+	}
+	return blueprints
+}
+
+func newWorkspaceFromV0(workspaceStruct workspaceV0) map[string]blueprint.Blueprint {
+	workspace := make(map[string]blueprint.Blueprint)
+	for name, blueprint := range workspaceStruct {
+		workspace[name] = blueprint.DeepCopy()
+	}
+	return workspace
+}
+
+func newComposesFromV0(composesStruct composesV0) map[uuid.UUID]compose.Compose {
+	composes := make(map[uuid.UUID]compose.Compose)
+
+	for composeID, composeStruct := range composesStruct {
+		c := compose.Compose{
+			Blueprint: composeStruct.Blueprint,
+		}
+		if len(composeStruct.ImageBuilds) == 0 {
+			panic("the was a compose with zero image builds, that is forbidden")
```

Comment:
`s|the|there|`?

Reply:
Thanks!

Reply:
This was dropped in a later patch, I'll leave the typo in the git history ;)